addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -2261,29 +2261,21 @@ var BasicModel = AbstractModel.extend({
      * @returns {Object}
      */
     _evalModifiers: function (element, modifiers) {
-        var result = {};
-        var self = this;
-        var evalContext;
-        function evalModifier(mod) {
-            if (mod === undefined || mod === false || mod === true) {
-                return !!mod;
-            }
-            evalContext = evalContext || self._getEvalContext(element);
-            return new Domain(mod, evalContext).compute(evalContext);
-        }
-        if ('invisible' in modifiers) {
-            result.invisible = evalModifier(modifiers.invisible);
-        }
-        if ('column_invisible' in modifiers) {
-            result.column_invisible = evalModifier(modifiers.column_invisible);
-        }
-        if ('readonly' in modifiers) {
-            result.readonly = evalModifier(modifiers.readonly);
-        }
-        if ('required' in modifiers) {
-            result.required = evalModifier(modifiers.required);
-        }
-        return result;
+        var evalContext = _.once(this._getEvalContext.bind(this, element));
+        return _(['invisible', 'column_invisible', 'readonly', 'required']).chain()
+            .filter(function (k) { return k in modifiers; })
+            .map(function (k) {
+                var mod = modifiers[k];
+                if (mod === undefined || mod === false || mod === true) {
+                    return [k, !!mod];
+                }
+                try {
+                    return [k, new Domain(mod, evalContext()).compute(evalContext())];
+                } catch (e) {
+                    throw new Error(_.str.sprintf('for modifier "%s": %s', k, e.message));
+                }
+            })
+            .compact().object().value();
```

Comment:
we try to restrict as much as possible the use of underscore, lol

Reply:
What version of JS/ES is allowed in non-test code on master?

Reply:
Also Object.fromEntries is super recent so sadness.

Reply:
You can use everything supported by all evergreen browsers, so unfortunately not Object.fromEntries as Edge doesn't implement it yet.